assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * ActivationBanner component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ReminderBanner from './ReminderBanner';
+import SetupBanner from './SetupBanner';
+import SuccessBanner from './SuccessBanner';
+import {
+	ACTIVATION_STEP_REMINDER,
+	ACTIVATION_STEP_SETUP,
+	ACTIVATION_STEP_SUCCESS,
+} from '../../../constants';
+
+export default function ActivationBanner() {
+	const [ step, setStep ] = useState( ACTIVATION_STEP_REMINDER );
+
+	const handleCTAClick = () => {
+		switch ( step ) {
+			case ACTIVATION_STEP_REMINDER:
+				setStep( ACTIVATION_STEP_SETUP );
+				break;
+			case ACTIVATION_STEP_SETUP:
+				setStep( ACTIVATION_STEP_SUCCESS );
+				break;
+		}
+		return { dismissOnCTAClick: false };
+	};
+
+	switch ( step ) {
+		case ACTIVATION_STEP_REMINDER:
+			return (
+				<ReminderBanner onCTAClick={ handleCTAClick }></ReminderBanner>
+			);
+		case ACTIVATION_STEP_SETUP:
+			return <SetupBanner onCTAClick={ handleCTAClick }></SetupBanner>;
+		case ACTIVATION_STEP_SUCCESS:
+			return <SuccessBanner />;
+		default:
+			return null;
+	}
```

Comment:
I'd probably nix the `switch` here too, personally, but I guess this one is more okay. It's still more indentation/nesting but at least with `return`s easy to follow. 🙂

Reply:
I personally find `switch`-es easier to read over `if`-s when there is a single conditional variable. But happy either way.  🙂

Reply:
Fair enough. I guess I am wary of the nesting, as usually nesting indicates "deeper logic", but in the case of switch statements one line is just for the variable being tested, and the comparison isn't all on one line.

But for this sort of return switch I'm less bothered 🙂